packages/es-components/src/components/containers/popoverTest/PopoverTest.js

```diff
@@ -0,0 +1,273 @@
+/* eslint-disable no-confusing-arrow */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { ThemeProvider } from 'styled-components';
+import viaTheme from 'es-components-via-theme';
+
+import { findDOMNode } from 'react-dom';
+
+import Icon from '../../base/icons/Icon';
+import Button from '../../controls/buttons/Button';
+
+import Popup from './Popup';
+
+const Container = styled.div`
+  display: inline-block;
+`;
+
+const TriggerButton = styled(Button)`
+  border-bottom: ${props => (props.suppressUnderline ? 'none' : '1px dotted')};
+  padding-bottom: ${props => (props.isLinkButton ? '3px' : '')};
+  text-decoration: none;
+
+  color: blue;
+`;
+
+const TriggerButtonLabel = styled.span`
+  font-size: 0;
+  height: 1px;
+  overflow: hidden;
+  display: block;
+`;
+
+const PopoverContainer = styled.div`
+  max-width: 275px;
+  background: ${props => props.theme.colors.white};
+  background-clip: padding-box;
+  border: 1px solid rgba(0, 0, 0, 0.2);
+  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);
+`;
+
+const PopoverHeader = styled.div`
+  display: flex;
+  justify-content: space-between;
+  color: ${props => props.theme.colors.white};
+  background-color: ${props =>
+    props.hasTitle ? props.theme.colors.popoverHeader : 'none'};
+`;
+
+const TitleBar = styled.h3`
+  font-size: 18px;
+  padding: 8px 14px;
+  margin: 0;
+`;
+
+const PopoverBody = styled.div`
+  padding: ${props =>
+    props.hasAltCloseWithNoTitle ? '0 14px 8px' : '8px 14px'};
+  text-align: right;
+`;
+
+const PopoverContent = styled.div`
+  text-align: left;
+  margin-bottom: ${props => (props.showCloseButton ? '8px' : '0')};
+`;
+
+const AlternateCloseButton = styled(Button)`
+  color: ${props =>
+    props.hasTitle ? props.theme.colors.white : props.theme.colors.grayDark};
+  margin: 0;
+  margin-left: auto;
+  padding: 0 8px;
+  background: transparent;
+  border: none;
+  box-shadow: none;
+
+  &:hover {
+    background: transparent;
+  }
+`;
+
+class PopoverTest extends React.Component {
+  state = {
+    isOpen: false
+  };
+
+  componentDidMount() {
+    this.header = findDOMNode(this.headerRef);
+    this.content = findDOMNode(this.contentRef);
+    this.closeBtn = findDOMNode(this.closeBtnRef);
+    this.triggerBtn = findDOMNode(this.triggerBtnRef);
+  }
+
+  toggleShow = () => {
+    const { isOpen } = this.state;
+
+    setTimeout(() => {
+      const focusableContent = this.content.querySelector('a, button');
+      if (focusableContent) {
+        focusableContent.focus();
```

Comment:
This makes me a little uncomfortable as well. Why are we moving the user's focus around? What is this buying us? I need to try using a keyboard, but this seems like it would lead to a weird experience with knowing where my tab is.

Reply:
We move focus because of keyboard accessibility standards. Things that pop up on the screen, like modals or non-modal dialogs should have their first navigable element or body content focused for keyboard users. And closing the modal or non-modal dialog should move the focus back to the calling element.